subsys/shell/shell.c

```diff
@@ -1131,7 +1131,8 @@ static void shell_log_process(const struct shell *shell)
 		 * readable and can be used to enter further commands.
 		 */
 		if (shell->ctx->cmd_buff_len) {
-			k_sleep(K_MSEC(15));
+			k_yield();
+			k_sleep(K_MSEC(1)); /* Sleep do not seem to yield. */
```

Comment:
k_sleep should yield if there is something to yield to. But if there is nothing to yield to nothing will seem to happen, except from time passing

Reply:
Yes it should, but it does not seem to do that for the native posix build.

Reply:
That should not be the case. So if you can create a case which reproduces it, I'd be very happy to see it.

Reply:
I found that the native_posix build sets the system to 32 bit, hence user_data size was 4 - which caused caused the ISO subsystem to do memory corruption when writing 10 bytes of data into this 4 byte buffer. The result in a clean build is that the data flow stops after 5-10 ISO packets. When I added the calls to yield it fixed the data flow, hence I concluded k_sleep() did not yield, where the explanation is more likely that the linker rearranged some memory when rebuilding the files I changed.
After fixing the user_data size the calls to yield is no longer needed, and I'll remove it again.